qiskit/result/utils.py

```diff
@@ -190,14 +187,7 @@ def _marginalize(counts, indices=None):
 
     # Sort the indices to keep in descending order
     # Since bitstrings have qubit-0 as least significant bit
-    indices = sorted(indices, reverse=True)
-
-    # Build the return list
-    new_counts = Counter()
-    for key, val in counts.items():
-        new_key = "".join([_remove_space_underscore(key)[-idx - 1] for idx in indices])
-        new_counts[new_key] += val
-    return dict(new_counts)
+    return results_rs.marginal_counts(counts, sorted(indices))
```

Comment:
I left the comment in place because it just applies to the inline `sorted(indices)` instead of doing `indices = sorted(indices, reverse=True)`. They're doing the same functional thing to preserve the independence of index of interest ordering from the output (basically don't permute the bits if you did something like `[1, 0]`).